main.go

```diff
@@ -103,3 +106,44 @@ func main() {
 			Version:       internal.FromBuild().Version,
 		})
 }
+
+func readApiKey(apiKeyStdin bool, cli command.Cli) (string, error) {
+	var apiKey string
+
+	if apiKeyStdin {
+		contents, err := io.ReadAll(cli.In())
+		if err != nil {
+			return "", err
+		}
+
+		apiKey = strings.TrimSuffix(string(contents), "\n")
+		apiKey = strings.TrimSuffix(apiKey, "\r")
+	} else if v, ok := os.LookupEnv("ATOMIST_API_KEY"); v != "" && ok {
```

Comment:
So, it looks like the `ATOMIST_API_KEY` env-var is only read when calling the `docker base login` command, which then saves it to the `~/.docker/config.json` file; I wonder if users would expect the env-var to be usable as an _alternative_ to saving the token in the  config file, i.e.


```bash
# export ATOMIST_API_KEY=<my key>
...
...
docker base detect <some image>
```


Or can the API key _only_ be used in combination with a workspace?

Reply:
Yeah, the `API_KEY` requires a `workspace` to be set as well. You need both.